interfaces/HO-Portal/src/app/programs/program-details/program-details.component.ts

```diff
@@ -1,32 +1,108 @@
 import { Component, OnInit } from '@angular/core';
-import { Router, ActivatedRoute } from '@angular/router';
+import { ActivatedRoute } from '@angular/router';
 import { ProgramsServiceApiService } from 'src/app/services/programs-service-api.service';
 import { Program } from 'src/app/models/program.model';
+import { ModalController } from '@ionic/angular';
+import { ProgramModalPage } from '../program-modal/program-modal.page';
+import { OverlayEventDetail } from '@ionic/core';
+import { TranslateService } from '@ngx-translate/core';
 
 @Component({
   selector: 'app-program-details',
   templateUrl: './program-details.component.html',
   styleUrls: ['./program-details.component.scss'],
 })
 export class ProgramDetailsComponent implements OnInit {
-  public program: any;
+  public languageCode: string;
+  public fallbackLanguageCode: string;
+
+  public program: Program;
+  public programTitle: string;
+  public programArray: any;
+
+  private techFeatures = [
+    'countryId',
+    'schemaId',
+    'credDefId',
+    'credOffer',
+    'proofRequest',
+  ];
+
 
   constructor(
     private route: ActivatedRoute,
-    private router: Router,
-    private programsService: ProgramsServiceApiService
-  ) { }
+    private programsService: ProgramsServiceApiService,
+    public modalController: ModalController,
+    public translate: TranslateService,
+  ) {
+
+  }
 
   ngOnInit() {
+    this.fallbackLanguageCode = this.translate.getDefaultLang();
+    this.languageCode = this.translate.currentLang;
+
     const id = this.route.snapshot.paramMap.get('id');
     this.programsService.getProgramById(id).subscribe((response) => {
-      this.program = this.generateArray(response);
-      console.log(this.program);
+      this.program = response;
+      this.programTitle = this.mapLabelByLanguageCode(this.program.title);
+      this.programArray = this.generateArray(this.program);
     });
   }
 
+
+
   public generateArray(obj) {
-    return Object.keys(obj).map((key) => ({ key, value: obj[key] }));
+    const result = [];
+    for (const key in obj) {
+      if (this.techFeatures.indexOf(key) <= -1) {
+
+        const keyNew = this.translate.instant('page.programs.program-details.' + key);
+        const valueNew = this.mapLabelByLanguageCode(obj[key]);
+
+        let isArray = false;
+        if (valueNew instanceof Array) {
+          if (typeof valueNew[0] === 'object') {
+            isArray = true;
```

Comment:
I think there are a few (helper-)methods within Angular/ES6 that could simplify this 3-levels-of-indentation..

Also, I think we don't really need to 'convert' the object in this way, but we can do a lot of it just in the template; By including some `*ngIf`-blocks and `*ngFor`-loops.

Because if we start adding more logic into this one method (like mentioned in the comment below), it will all become a big tangled and indented pile of code quickly.

If we define methods to do small (generic) operations, we can re-use them in all the different places we need them. All the 'specifics' will/should be contained in an 'init' or 'setup/prepare'-method only;

Reply:
Sure, but I'll leave this until it's actually worked on. Right now it's not used.